thirdeye/thirdeye-frontend/app/pods/components/alert-report-modal/component.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Component for "report missing anomaly" modal
+ * @module components/alert-report-modal
+ * @property {String} metricName - text for read-only metric field
+ * @property {String} alertName  - text for read-only alert field
+ * @property {Array} dimensionOptions - options for dimension select field
+ * @property {Number} timePickerIncrement - config for time-range-picker
+ * @property {String} maxTime - timestamp for loading anomaly graph
+ * @property {String} viewRegionStart - range start timestamp
+ * @property {String} viewRegionEnd - range end timestamp
+ * @property {Object} predefinedRanges - needed for time-range-picker
+ * @property {String} uiDateFormat - date format desired for time-range-picker
+ * @property {String} graphMessageText - text for graph in loading state
+ * @example
+  {{alert-report-modal
+    metricName="mobile_notification_errors"
+    alertName="notification_sessions_mobile"
+    dimensionOptions=['dimension 1', 'dimension 2']
+    timePickerIncrement=200
+    maxTime="1513137100914"
+    viewRegionStart="2017-10-12 23:59"
+    viewRegionEnd="2017-12-11 23:59"
+    predefinedRanges=predefinedRanges
+    uiDateFormat="MMM D, YYYY hh:mm a"
+    graphMessageText="Loading graph"
+    inputAction=(action "onInputMissingAnomaly")
+  }}
+ * @exports alert-report-modal
+ * @author smcclung
+ */
+
+import Ember from 'ember';
+import _ from 'lodash';
+
+export default Ember.Component.extend({
+  containerClassNames: 'alert-report-modal',
+  isNewTrend: false,
+
+  /**
+   * Collects all input data for the post request
+   * @method reportAnomalyPayload
+   * @return {Object} Post data
+   */
+  reportAnomalyPayload: Ember.computed(
+    'isNewTrend',
+    'anomalyComments',
+    'selectedDimension',
+    'viewAnomalyStart',
+    'viewAnomalyEnd',
+    'anomalyLinks',
+    function() {
+      const postObj = {
+        startTime: this.get('viewAnomalyStart'),
+        endTime: this.get('viewAnomalyEnd'),
+        feedbackType: this.get('isNewTrend') ? 'ANOMALY_NEW_TREND' : 'ANOMALY',
+        dimension: this.get('selectedDimension') || null,
+        comment: this.get('anomalyComments') || null,
+        externalURL: this.get('anomalyLinks') || null
+      };
+
+      return postObj;
+    }
+  ),
+
+  /**
+   * Sends post object as is to parent
+   * @method bubbleModalInput
+   */
+  bubbleModalInput() {
+    this.sendAction('inputAction', this.get('reportAnomalyPayload'));
```

Comment:
fyi, I personally prefer passing a closure action into the component and call it in the action `this.attrs.inputAction()`